client/src/featureform/local_cache.py

```diff
@@ -181,11 +181,19 @@ def get_source_files_for_source(self, source_name, source_variant) -> Set[str]:
             dependencies = json.loads(source["inputs"])
             for name, variant in dependencies:
                 sources.update(self.get_source_files_for_source(name, variant))
+        elif transform_type == SourceType.DIRECTORY.value:
+            path = source["definition"]
+            for absolute_file, _ in self._absolute_file_paths(path):
+                sources.update([absolute_file])
         else:
             raise Exception(f"Unknown source type: {transform_type}")
-
         return sources
 
+    def _absolute_file_paths(self, directory):
+        for dirpath, _, filenames in os.walk(directory):
+            for f in filenames:
+                yield os.path.abspath(os.path.join(dirpath, f)), f
```

Comment:
nit: probz don't need the the second return var

Reply:
changed